### PR TITLE
Add export-import related JS best practices

### PR DESCRIPTION
**Pull request by thisismydesign**

### CODE REVIEW

File: js/best-practices.md [hunk 1]
```diff
@@ -1,5 +1,58 @@
...
+Instead, prefer breaking up the code into separate default exported modules or use named exports. See also:
+- https://geedew.com/es6-module-gotchas/
+- https://github.com/airbnb/javascript/issues/710#issuecomment-297840604
+- https://2ality.com/2014/09/es6-modules-final.html#default-exports-are-favored
+
+BAD
+
+```js
+export default { a, b, c };
+```
```
Comment (gomorizsolt) [inline js/best-practices.md]: Could you add a "GOOD" example here?

**Comment by thisismydesign** [comment]
> Would it make sense to add https://exploringjs.com/es6/ch_modules.html to the resources? It's a great introductory to this whole module-system imo.

I had that initially actually but found it way too broad for these items. It is a good overview, but I don't know where to include it. I would like that reading through the resources are directly helpful for these rules, but this one mostly gives a general understanding. It's also linked within one of the sources I believe.

File: js/best-practices.md (hunk 1 above)
Comment (thisismydesign) [reply js/best-practices.md]: https://github.com/c-hive/guides/pull/38/commits/170d514168f22a5c8a58d898ec7f925f4952bb1b

Comment (gomorizsolt) [reply js/best-practices.md]: Thanks!

**Pull request merged**